test/lib/ex_aws/config_test.exs

```diff
@@ -5,7 +5,7 @@ defmodule ExAws.ConfigTest do
     value = "foo"
     System.put_env("ExAwsConfigTest", value)
     assert :s3
-    |> ExAws.Config.new([access_key_id: {:system, "ExAwsConfigTest"}])
+    |> ExAws.Config.new([access_key_id: {:system, "ExAwsConfigTest"}, secret_access_key: {:system, "AWS_SECURITY_TOKEN"}])
```

Comment:
It appears that if you don't overwrite this, the default way of loading secret_access key will be used (and I updated the defaults to include `:awscli`) -- and the way the Config module works, it loads all variables the config options provide -- overwriting other settings. Does this make sense?

Likewise, :instance_profile will override a few keys as per https://github.com/tomasz-tomczyk/ex_aws/blob/bceaf3bc55bda27df995b778629d35d1ec3d2695/lib/ex_aws/config.ex#L64-L68

In theory, it should be possible to load access_key_id from ENV and secret_access_key from another source, but doesn't seem possible by design with current implementation. Should be an edge case, anyway, so not sure if worth refactoring?